src/CMakeLists.txt

```diff
@@ -79,6 +80,7 @@ SET(core_headers
 
 PODIO_ADD_LIB_AND_DICT(podio "${core_headers}" "${core_sources}" selection.xml)
 target_compile_options(podio PRIVATE -pthread)
+target_link_libraries(podio PRIVATE Python3::Python)
```

Comment:
Ah, this probably should have been in the `Python::` namespace since that's what is added as a dependency in the config
https://github.com/AIDASoft/podio/blob/5d827486312154a0b6e2cb095ec28371d6ee5220/cmake/podioConfig.cmake.in#L28-L32
and in the `PODIO_GENERATE_DATAMODEL` module:
https://github.com/AIDASoft/podio/blob/5d827486312154a0b6e2cb095ec28371d6ee5220/cmake/podioMacros.cmake#L158